File: app/src/main/java/ke/choxxy/wordsearch/features/mainmenu/MainMenuViewModel.java
package ke.choxxy.wordsearch.features.mainmenu;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import ke.choxxy.wordsearch.data.GameThemeRepository;
import ke.choxxy.wordsearch.model.GameTheme;

import java.util.List;

public class MainMenuViewModel extends ViewModel {

    private GameThemeRepository mGameThemeRepository;

    private MutableLiveData<List<GameTheme>> mOnGameThemeLoaded;

    public MainMenuViewModel(GameThemeRepository gameThemeRepository) {
        mGameThemeRepository = gameThemeRepository;
        mOnGameThemeLoaded = new MutableLiveData<>();
    }

    public void loadData() {
        mOnGameThemeLoaded.setValue(mGameThemeRepository.getGameThemes());
    }

    public LiveData<List<GameTheme>> getOnGameThemeLoaded() {
        return mOnGameThemeLoaded;
    }
}

File: app/src/main/java/ke/choxxy/wordsearch/mainmenu/MainMenuViewModel.kt
package ke.choxxy.wordsearch.mainmenu

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import dagger.hilt.android.lifecycle.HiltViewModel
import ke.choxxy.wordsearch.data.GameThemeRepository
import ke.choxxy.wordsearch.model.GameTheme
import javax.inject.Inject

@HiltViewModel
class MainMenuViewModel @Inject constructor(private val mGameThemeRepository: GameThemeRepository) : ViewModel() {

    private val mOnGameThemeLoaded: MutableLiveData<List<GameTheme>> = MutableLiveData()

    fun loadData() {
        mOnGameThemeLoaded.value = mGameThemeRepository.gameThemes
    }

    val onGameThemeLoaded: LiveData<List<GameTheme>>
        get() = mOnGameThemeLoaded
}
